torch/csrc/jit/passes/erase_number_types.cpp

```diff
@@ -6,13 +6,19 @@
 namespace torch {
 namespace jit {
 
+void SetNumTypeToTensorType(Value* v) {
+  if (v->type()->isSubtypeOf(NumberType::get())) {
+    v->setType(TensorType::fromNumberType(v->type()));
+  } else if (v->type()->isSubtypeOf(BoolType::get())) {
+    v->setType(TensorType::fromBoolType());
+  }
+}
+
 void EraseNumberTypesOnBlock(Block* block) {
   for (auto it = block->nodes().begin(), end = block->nodes().end(); it != end;
        ++it) {
     for (auto inp : it->inputs()) {
-      if (inp->type()->isSubtypeOf(NumberType::get())) {
-        inp->setType(TensorType::get());
-      }
+      SetNumTypeToTensorType(inp);
```

Comment:
We originally only deal with NumberType not BoolType, now it considers BoolType as well. Is this on purpose?